wagtail/search/backends/db.py

```diff
@@ -51,36 +51,45 @@ def build_single_term_filter(self, term):
             term_query |= models.Q(**{field_name + '__icontains': term})
         return term_query
 
-    def check_boost(self, query):
-        if query.boost != 1:
+    def check_boost(self, query, boost=1.0):
+        if query.boost * boost != 1.0:
             warn('Database search backend does not support term boosting.')
 
-    def build_database_filter(self, query=None):
+    def build_database_filter(self, query=None, boost=1.0):
         if query is None:
             query = self.query
 
+        if isinstance(query, PlainText):
+            self.check_boost(query, boost=boost)
+
+            operator = {
+                'and': AND,
+                'or': OR,
+            }[query.operator]
```

Comment:
That dictionary could be moved as a class attribute for performance and consistency with `DEFAULT_OPERATOR`.